### ISSUE TITLE
Drag the browser window around block the main thread

### ISSUE THREAD

**Issue by lanigela** [electron/electron#8710]
I'm developing a computationally intensive application using electron. I came to notice that when I drag the browser around, SOMETIMES the main Node.js thread is blocked, and it blocks the main thread for a while, or even completely by change. 

I first thought it might be the communication between IPCMain and IPCRender so I disconnect all of them in my code, but it doesn't help. This is like a reverse case of #3363, however as far as I know I can't workaround it not touching code in electron.

To make a simple illustration, I use the electron quick start application and 15 lines additional code.

* Electron version: 1.4.15
* Operating system: Windows 10 Pro Version 10.0.14393 (Also reproduce-able on Ubuntu )

### How to reproduce

1. Download electron-quick-start
https://github.com/electron/electron-quick-start
2. In main.js add
```sh
var lastDate = new Date();

function printTime () {
  var d = new Date();
  var delta = d - lastDate;
  if(delta > 20) {
    console.log(delta);
  }
  lastDate = d;
}
```
and modity app on ready
```sh
app.on('ready', () => {
  createWindow();

  setInterval(printTime, 10);
})
```
3. Run the app, and drag the title bar of the browser window around 

### Expected behavior

If the main thread keeps running, the interval calls printTime every 10ms and the threshold for the console.log is 20ms, so there should be no print out in the console.

### Actual behavior
I get random numbers like
```sh
96
34
33
601
118
32
```

**Comment by bhouston** [comment]
This is a pretty serious issue for us.  Basically we have a "real-time" application running in Node.JS and Electron is causing it to halt for arbitrary long periods if someone moves the window around.  This is not an issue with standard GUI applications on Windows or Ubuntu, rather this limitation of locking the main loop when the window is being moved is imposed by either Chromium or Electron.

**Comment by groundwater** [comment]
Hey @lanigela, thanks for posting the detailed repro-case.

I know we're looking at node-integration into web-workers, which can run in their own threads. If you or @bhouston have any more information about your applications, we could use that as a user-story when working on the feature.

We can also have a look and see if this problem is avoidable, or too baked into the architecture to avoid without using other threads.

**Comment by bhouston** [comment]
@groundwater Thanks for the attention!  Wohoo!

I can describe the app at a high level.  It is a near real-time signal processing tool that uses a lot of CUDA via Node.JS NAN plugin and has a 3D front end and React.js-based GUI.  Basically electron allows for a cross platform application with a rich UI at very low cost -- basically a replacement for Qt's normal role.  We love electron and it has saved us a ton of money in development.  It helps that we are a web-focused development shop with lots of web front-end experience.  While our app doesn't have hard real-time constraints like a true real-time control system, it does output processed sound in a real-time nature.  This sound is composed of a lot of short segments (as they come out of the real-time processing system) that if we have near real-time behavior out of the app, it keeps the output audio buffer from going empty and thus the sound appears continuous.  This mostly works until the user decides to move the window with the React.JS GUI in it and then there is a good chance (but not fully consistent -- it seems to increase its consistency the more stuff the main thread is doing) the sound and display (which is also feed real-time) stops updating because the main loop blocked on something.  This blocking continues until they release their mouse cursor or stop moving the window -- I am a little confused on exactly what triggers the end of the block.

Another clue: We know that Chromium will keep playing a Three.JS animation even when moving or resizing the window on Ubuntu and Windows.  Thus this doesn't block Chromium.

**Comment by zcbenz** [comment]
This is very likely a duplicate of #7079.
